configserver/src/test/java/com/yahoo/vespa/config/server/deploy/HostedDeployNodeAllocationTest.java

```diff
@@ -0,0 +1,119 @@
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void testDeployMultipleVersionsCreatingDifferentHosts() {
+        List<ModelFactory> modelFactories = List.of(createHostedModelFactory(Version.fromString("7.2")),
+                                                    createHostedModelFactory(Version.fromString("7.3")));
+        var provisioner = new VersionProvisioner();
+        DeployTester tester = new DeployTester.Builder(temporaryFolder).modelFactories(modelFactories)
+                                                                       .provisioner(new MockProvisioner().hostProvisioner(provisioner))
+                                                                       .hostedConfigserverConfig(Zone.defaultZone())
+                                                                       .build();
+        tester.deployApp("src/test/apps/hosted/", "7.3");
+
+        var hosts = containers(tester.getAllocatedHostsOf(tester.applicationId()).getHosts());
+        assertEquals("Allocating the superset of hosts of each version", 5, hosts.size());
+        assertEquals(resources(3), get("host0", hosts).advertisedResources());
+        assertEquals(resources(3), get("host1", hosts).advertisedResources());
+        assertEquals(resources(3), get("host2", hosts).advertisedResources());
+        assertEquals(resources(2), get("host3", hosts).advertisedResources());
+        assertEquals(resources(2), get("host4", hosts).advertisedResources());
+    }
+
+    private HostSpec get(String hostname, Set<HostSpec> hosts) {
+        return hosts.stream().filter(host -> host.hostname().equals(hostname)).findAny().orElseThrow();
+    }
+
+    private Set<HostSpec> containers(Set<HostSpec> hosts) {
+        return hosts.stream().filter(host -> host.membership().get().cluster().type() == ClusterSpec.Type.container).collect(Collectors.toSet());
+    }
+
+    private static NodeResources resources(double vcpu) {
+        return new NodeResources(vcpu, 1, 1, 1);
+    }
+
+    private static class VersionProvisioner implements HostProvisioner {
+
+        int invocation = 0;
+
+        @Override
+        public HostSpec allocateHost(String alias) {
+            throw new RuntimeException();
+        }
+
+        @Override
+        public List<HostSpec> prepare(ClusterSpec cluster, Capacity capacity, ProvisionLogger logger) {
+            if (cluster.id().value().equals("container")) { // the container cluster from the app package: Use this to test
+                if (invocation == 0) { // Building the latest model version, 7.3: Always first
+                    invocation++;
+                    // Returns nodes which are on both versions to trigger building old versions
+                    return List.of(host("host0", resources(3), 0, "7.3", cluster),
+                                   host("host1", resources(3), 1, "7.3", cluster),
+                                   host("host2", resources(3), 2, "7.2", cluster));
```

Comment:
```suggestion
                                   host("host2", resources(3), 2, "7.3", cluster));
```

Reply:
It must be like this actually - this isn't the model version but the node's version, and these versions decide what model versions we'll build.